kolena/_experimental/dataset/_dataset.py

```diff
@@ -113,15 +121,52 @@ def _to_deserialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     return result
 
 
-def register_dataset(name: str, df: pd.DataFrame) -> None:
+def _upload_dataset_chunk(df: pd.DataFrame, load_uuid: str, id_fields: List[str]) -> None:
+    df_serialized_datapoint = _to_serialized_dataframe(df, column=COL_DATAPOINT)
+    df_serialized_datapoint_id_object = _to_serialized_dataframe(df[sorted(id_fields)], column=COL_DATAPOINT_ID_OBJECT)
+    df_serialized = pd.concat([df_serialized_datapoint, df_serialized_datapoint_id_object], axis=1)
+
+    upload_data_frame(df=df_serialized, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
+
+
+def load_dataset(name: str) -> Optional[EntityData]:
+    response = krequests.put(Path.LOAD_DATASET, json=asdict(LoadDatasetByNameRequest(name=name)))
+    if response.status_code == requests.codes.not_found:
+        return None
+
+    response.raise_for_status()
+
+    return from_dict(EntityData, response.json())
+
+
+def register_dataset(name: str, df: Union[Iterator[pd.DataFrame], pd.DataFrame], id_fields: List[str]) -> None:
     """
-    Create or update a dataset with datapoints.
+    Create or update a dataset with datapoints and id_fields. If the dataset already exists, in order to associate the
+    existing result with the new datapoints, the id_fields need be the same as the existing dataset.
+
+    :param name: name of the dataset
+    :param df: an iterator of pandas dataframe or a pandas dataframe, you can pass in the iterator if you want to have
+                batch processing,
+                 example iterator usage: csv_reader = pd.read_csv("PathToDataset.csv", chunksize=10)
+    :param id_fields: a list of id fields, this will be used to link the result with the datapoints
```

Comment:
@ly-kolena thoughts on inferring these? For the simple case, I'd like for users to not have to think about ID.

Reply:
The discussion was

1. On top of existing datapoint_type infer, we can also auto-configure for dataframe with either `locator` or `text` column to use them. respectively, and raise exception when neither is found and user does not provide `id_fields`.
2. Warn users if the final resolved `id_fields` differ from existing one of the dataset (if this is an update)
3. (Optional future feature) augment handling of 2) to prompt for user confirmation, with option like `force=True` to bypass prompt, in the future for better visibility on `id_fields` change.

One downside is that, `id_fields` is still critical for "consistent" data handling, e.g. datapoint and their results. It would still come as a surprise if users are not aware of it in the first place and then have to come back to comb through documentation to realize this is the cause of data ingestion error, or something else.

Does that sound reasonable?

Reply:
@grant-Kolena , have you addressed this concern?

Reply:
We think the infer should happen on galleon side, if someone is integrating with SDK they should provide the id explicitly as there is no interface to let them select /inform them what has been selected as id field.

Also I am not clear on the rules of what should be inferred as id

Reply:
discussed offline, just inferring locator and text for now